[Refactoring] Convert completion handler when converting function to async

ahoppen opened the pull request:
Convert function to async currently only adds "async" to the function and runs the convert call refactoring on the body.

This was intentional, but it turns out to be somewhat confusing. Instead, run the same refactoring as the add async alternative refactoring but just replace rather than add.

ahoppen commented:
@swift-ci Please test

bnbarham commented on test/refactoring/ConvertAsync/basic.swift:
```diff
@@ -151,14 +151,9 @@ func retStruct() -> MyStruct { return MyStruct() }
 protocol MyProtocol {
   // RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+2):3 | %FileCheck -check-prefix=PROTO-MEMBER %s
-  // RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=PROTO-MEMBER-TO-ASYNC %s
+  // RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=PROTO-MEMBER %s
```
Is there a test that checks that it does actually *convert* and not *add* anywhere? See the top of this file for an example (it's for add-async, convert should have an actual range rather than a 0 length one).

ahoppen replied:
I don’t think so. I’ll add one in a follow-up PR.

bnbarham reviewed (approved):
LGTM otherwise, thanks!

ahoppen commented:
@swift-ci Please test Windows

ahoppen committed:
[Refactoring] Convert completion handler when converting function to async

Convert function to async currently only adds "async" to the function and runs the convert call refactoring on the body.

This was intentional, but it turns out to be somewhat confusing. Instead, run the same refactoring as the add async alternative refactoring but just replace rather than add.

Resolves rdar://77103049

ahoppen commented:
@swift-ci Please test

ahoppen commented:
Windows failures is unrelated.
